### PR TITLE
Add Insiders tag on profile page

### PR DESCRIPTION
**Pull request by dem4ron**
Screenshot 2023-04-13 at 13 02 14

### CODE REVIEW

File: app/views/profiles/_header.html.haml [hunk 1]
```diff
@@ -45,16 +45,18 @@
...
-                  .title Exercism Staff
-              - if user.maintainer?
-                .tag.maintainer
-                  = graphical_icon :maintaining
-                  .title Maintainer
+              - tags = []
+              - tags << { class: "tag staff", icon: :logo, title: "Exercism Staff" } if user.staff?
+              - tags << { class: "tag maintainer", icon: :maintaining, title: "Maintainer" } if user.maintainer?
+              -# TODO: add user.insider?
+              - tags << { class: "tag maintainer", icon: :insiders, title: "Insider" } if user.maintainer?
```
Comment (iHiD) [inline app/views/profiles/_header.html.haml]: Could you make this a draft until this is done pls? So we don't accidently merge?

File: app/helpers/view_components/profile/header.rb [hunk 2]
```diff
@@ -23,6 +24,14 @@ def top_three_tracks
         ::Track.where(id: track_ids).
           order(Arel.sql("FIND_IN_SET(id, '#{track_ids.join(',')}')"))
       end
+
+      def header_tags
+        tags = []
+        tags << { class: "tag staff", icon: :logo, title: "Exercism Staff" } if @user.staff?
+        tags << { class: "tag maintainer", icon: :maintaining, title: "Maintainer" } if @user.maintainer?
+        tags << { class: "tag maintainer", icon: :insiders, title: "Insider" } if %i[eligible eligible_lifetime].include?(@user.insiders_status) # rubocop:disable Layout/LineLength
```
Comment (iHiD) [inline app/helpers/view_components/profile/header.rb]:
```suggestion
        tags << { class: "tag maintainer", icon: :insiders, title: "Insider" } if %i[active active_lifetime].include?(@user.insiders_status) # rubocop:disable Layout/LineLength
```

---

**Please also add a test for this.**

File: app/helpers/view_components/profile/header.rb [hunk 3]
```diff
@@ -23,6 +24,14 @@ def top_three_tracks
         ::Track.where(id: track_ids).
           order(Arel.sql("FIND_IN_SET(id, '#{track_ids.join(',')}')"))
       end
+
+      def header_tags
+        tags = []
+        tags << { class: "tag staff", icon: :logo, title: "Exercism Staff" } if @user.staff?
+        tags << { class: "tag maintainer", icon: :maintaining, title: "Maintainer" } if @user.maintainer?
+        tags << { class: "tag insider", icon: :insiders, title: "Insider" } if @user.insider?
```
Comment (ErikSchierboom) [inline app/helpers/view_components/profile/header.rb]: @iHiD Is this the right order? As in: do maintainer and staff tags have precedence over insider?

**Review by ErikSchierboom — APPROVED** [review]

File: app/helpers/view_components/profile/header.rb (hunk 3 above)
Comment (ErikSchierboom) [reply app/helpers/view_components/profile/header.rb]: Do we want to feature flag this?

Comment (iHiD) [reply app/helpers/view_components/profile/header.rb]: > @iHiD Is this the right order? As in: do maintainer and staff tags have precedence over insider?

Yes

> Do we want to feature flag this?

Yes for today.

Comment (ErikSchierboom) [reply app/helpers/view_components/profile/header.rb]: Okay, I added the feature flag

File: app/models/user.rb [hunk 4]
```diff
@@ -297,6 +297,7 @@ def profile? = profile.present?
   def confirmed? = super && !disabled? && !blocked?
   def disabled? = !!disabled_at
   def blocked? = User::BlockDomain.blocked?(user: self)
+  def insider? = insiders_status_active? || insiders_status_active_lifetime?
```
Comment (ErikSchierboom) [inline app/models/user.rb:300]: @iHiD You'll want to memoize this too in your branch

**Pull request merged**